src/core/globalstate.ts

```diff
@@ -88,7 +88,34 @@ export class MobXGlobals {
 
 export let globalState: MobXGlobals = new MobXGlobals();
 
+let shareGlobalStateCalled = false;
+let warnedAboutMultipleInstances = false;
+let _runInSandbox = false;
+
+{
+	const global = getGlobal();
+	if (!global.__mobxInstanceCount) {
+		global.__mobxInstanceCount = 1;
+	} else {
+		global.__mobxInstanceCount++;
+		setTimeout(() => {
+			if (!shareGlobalStateCalled && !warnedAboutMultipleInstances && !_runInSandbox) {
+				warnedAboutMultipleInstances = true;
+				console.warn("[mobx] Warning: there are multiple mobx active instances active. This might lead to unexpected results. See https://github.com/mobxjs/mobx/issues/1082 for details.")
+			}
+		})
+	}
+}
+
+export function runInSandbox() {
```

Comment:
Maybe reconsider the name, there is no concept of actual sandbox and there is `runInAction`, so someone may (I would) expect something more like this: 
```javascript
runInSandbox(() => {
  // runs isolated
} /*, sandbox/global/namespace/whatever*/)
```
`supressMultipleInstancesWarning` perhaps?